storagetransfer/api/StorageTransfer.Samples.Tests/StorageFixture.cs

```diff
@@ -0,0 +1,82 @@
+{
+    [CollectionDefinition(nameof(StorageFixture))]
+    public class StorageFixture : IDisposable, ICollectionFixture<StorageFixture>
+    {
+        public string ProjectId { get; }
+        public string BucketNameSource { get; } = Guid.NewGuid().ToString();
+        public string BucketNameSink { get; } = Guid.NewGuid().ToString();
+        public StorageClient Storage { get; } = StorageClient.Create();
+        public StorageTransferServiceClient Sts { get; } = StorageTransferServiceClient.Create();
+
+        public StorageFixture()
+        {
+            ProjectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+            if (string.IsNullOrWhiteSpace(ProjectId))
+            {
+                throw new Exception("You need to set the Environment variable 'GOOGLE_PROJECT_ID' with your Google Cloud Project's project id.");
+            }
+
+            CreateBucketAndGrantStsPermissions(BucketNameSink);
+            CreateBucketAndGrantStsPermissions(BucketNameSource);
+        }
+
+        private void CreateBucketAndGrantStsPermissions(string bucketName)
+        {
+            var bucket = Storage.CreateBucket(ProjectId, new Bucket
+            {
+                Name = bucketName
+            });
+
+            string email = Sts.GetGoogleServiceAccount(new GetGoogleServiceAccountRequest()
+            {
+                ProjectId = ProjectId
+            }).AccountEmail;
+            string member = "serviceAccount:" + email;
+            string objectViewer = "roles/storage.objectViewer";
+            string bucketReader = "roles/storage.legacyBucketReader";
+            string bucketWriter = "roles/storage.legacyBucketWriter";
+
+            var policy = Storage.GetBucketIamPolicy(bucketName, new GetBucketIamPolicyOptions
+            {
+                RequestedPolicyVersion = 3
+            });
+            // Set the policy schema version. For more information, please refer to https://cloud.google.com/iam/docs/policies#versions.
+            policy.Version = 3;
+
+            Policy.BindingsData objectViewerBinding = new Policy.BindingsData
+            {
+                Role = objectViewer,
+                Members = new List<string> { member }
+            };
+            Policy.BindingsData bucketReaderBinding = new Policy.BindingsData
+            {
+                Role = bucketReader,
+                Members = new List<string> { member }
+            };
+            Policy.BindingsData bucketWriterBinding = new Policy.BindingsData
+            {
+                Role = bucketWriter,
+                Members = new List<string> { member }
+            };
+
+            policy.Bindings.Add(objectViewerBinding);
+            policy.Bindings.Add(bucketReaderBinding);
+            policy.Bindings.Add(bucketWriterBinding);
+
+            Storage.SetBucketIamPolicy(bucketName, policy);
+        }
+
+        public void Dispose()
+        {
+            Storage.DeleteBucket(BucketNameSink);
```

Comment:
try ... catch, we delete resources on a best effor basis.

I usually add a SafeDeleteMethod and call it twice.